client/trino-jdbc/src/main/java/io/trino/jdbc/AbstractTrinoResultSet.java

```diff
@@ -156,6 +157,8 @@ abstract class AbstractTrinoResultSet
                     .add("date", String.class, java.time.LocalDate.class, string -> parseDate(string, CURRENT_TIME_ZONE, CURRENT_JAVA_TIME_ZONE).toLocalDate())
                     .add("time", String.class, Time.class, string -> parseTime(string, SYSTEM_DEFAULT_ZONE_ID))
                     .add("time with time zone", String.class, Time.class, AbstractTrinoResultSet::parseTimeWithTimeZone)
+                    .add("timestamp", String.class, Instant.class, string -> parseTimestampAsInstant(string, SYSTEM_DEFAULT_ZONE_ID))
+                    .add("timestamp", String.class, LocalDateTime.class, string -> parseTimestampAsLocalDateTime(string, SYSTEM_DEFAULT_ZONE_ID))
```

Comment:
Timestamp type in SQL doesn't have a time zone. Why does is parseTimestampAsLocalDateTime expecting one?

Reply:
Because the code was written to convert from String -> Instant (in system zone) -> LDT (see `toLocalDateTime` method in this class).

But you're right this could directly go from String -> LDT. Good catch, sorry I missed this.